externs/es6.js

```diff
@@ -2035,6 +2035,31 @@ function WeakRef(value) {}
  */
 WeakRef.prototype.deref = function() {};
 
+/**
+ * @constructor
+ * @struct
+ * @param {function(HOLDINGS)} cleanupCallback
+ * @template TARGET, HOLDINGS, TOKEN
+ * @nosideeffects
```

Comment:
I could not find any information on whether these could be side-effectful. Just out of curiosity, did you?

Reply:
As I understand it (please correct me if I am wrong), the `@nosideffects` declaration here just means that the call the the constructor can be omitted if the returned value is not used.  That appears to be the case: [the `FinalizationRegistry` constructor](https://tc39.es/ecma262/#sec-finalization-registry-cleanup-callback)  only really does three things:

* It calls [OrdinaryCreateFromConstructor](https://tc39.es/ecma262/#sec-ordinarycreatefromconstructor), which is called by basically every constructor in the spec and so I presume is regarded as side-effect free.
* It calls [HostMakeJobCallback](https://tc39.es/ecma262/#sec-hostmakejobcallback)(_cleanupCallback_), and
* It initialises a few slots on the new instance, including one with the result of the HostMakeJobCallback call.

HostMakeJobCallback appears to be somewhat loosely specified.  The default implementation is clearly side-effect free, but it's permissible that a web browser could implement it differently. There does not appear to be any guarantee that such an alternative implementation is strictly side-effect free, but this possibility seems sufficiently remote that it would seem unreasonable for compiler users to depend on a side effect of calling `void new FinalizationRegistry()`.